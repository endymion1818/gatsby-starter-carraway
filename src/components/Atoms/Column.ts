import styled from 'styled-components'
import { flexalign, size } from '../tokens'

export interface IColumnProps {
  verticalAlign?: string
  bufferTop?: string
  bufferBottom?: string
  textAlign?: string
}

const Column = styled.div<IColumnProps>`
    display: flex;
    flex-direction: column;

    align-self: ${({ verticalAlign = flexalign.start }) => verticalAlign};
    ${({ bufferTop = size.single }) => bufferTop && `margin-top: ${bufferTop};`}
    ${({ bufferBottom = size.single }) => bufferBottom && `margin-bottom: ${bufferBottom};`}

    ${({ textAlign }) => textAlign && `text-align: ${textAlign};`}

    > h1,
    > h2,
    > h3,
    > h4,
    > h5,
    > h6 {
        flex: 1;
    }
    > p {
        flex: 2;

        & ~ div[class*='Button'] {
        flex: 0;
        }
    }
    > img {
        width: 100%;
    }
`

export default Column
