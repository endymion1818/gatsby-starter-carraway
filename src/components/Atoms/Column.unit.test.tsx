import { cleanup, render } from '@testing-library/react'
import React from 'react'
import Column from './Column'

describe('<Column />', () => {
  beforeEach(cleanup)
  it('Should render', () => {
    const { container } = render(<Column />)
    expect(container).toMatchSnapshot()
  })
})
