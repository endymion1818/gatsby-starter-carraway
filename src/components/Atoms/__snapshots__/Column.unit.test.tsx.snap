// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Column /> Should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c0 > h1,
.c0 > h2,
.c0 > h3,
.c0 > h4,
.c0 > h5,
.c0 > h6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 > p {
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
}

.c0 > p ~ div[class*='Button'] {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
}

.c0 > img {
  width: 100%;
}

<div>
  <div
    class="c0"
  />
</div>
`;
